### ISSUE TITLE
Gradient checks across 5 backward suites silently degrade to an absolute bound; GQA backward correctness is unverified

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#998]
## Summary

The finite-difference gradient-check helper used across attention and SSM backward tests floors its denominator at `1.0`, which converts a stated **relative** tolerance into an **absolute** one for every gradient element smaller than 1.0 — which is all of them at these fixtures' scaling. The result is that gradient correctness for those layers is not actually being verified.

Found by the T152.3 vacuous-assertion audit (PR #997) and independently re-verified.

## The construction

`layers/attention/gqa_backward_test.go:321-325`:

```go
func gradClose(a, b, tol float32) bool {
    diff := float32(math.Abs(float64(a - b)))
    denom := float32(math.Max(1.0, math.Max(math.Abs(float64(a)), math.Abs(float64(b)))))
    return diff/denom < tol
}
```

When `|a|,|b| < 1`, `denom == 1.0` and the test reduces to `diff < tol`. With `tol = 1e-2` and gradients of order 1e-3, an error of **100% of the true value** passes.

## Blast radius: five suites, not one

The identical floored-denominator construction appears in:

- `layers/attention/gqa_backward_test.go:323`
- `layers/attention/sdpa_backward_finitediff_test.go:177`
- `layers/attention/mla_backward_test.go:170`
- `layers/ssm/bc_norm_test.go:202`
- `layers/ssm/mamba_block_test.go:287`

So grouped-query attention, scaled-dot-product attention, multi-head latent attention, BC-norm and Mamba block backward passes are all checked this way.

## Red-proof (this is the part that matters)

The audit **deleted the entire dK path** from GQA's input gradient computation. Nothing caught it: not `TestGQABackward_MHA`, not `TestGQABackward_WeightGradients`, not `tests/parity`, not `layers/transformer`. The whole suite stayed green with a backward pass that is missing one of its three input gradients.

## Compounding factor

`layers/attention/gqa_backward_test.go:17` — `TestGQABackward`, the only test covering the genuinely grouped case (`numQ != numKV`, i.e. actual GQA rather than MHA), is skipped:

```
t.Skip("Known gradient mismatch between analytical and numerical (finite-difference) backward pass. See docs/devlog.md.")
```

So there is a **known** gradient mismatch, the test that would catch it is disabled, and the remaining tests are too loose to catch it independently.

## What this means for product claims

Backward-pass correctness for every GQA architecture is currently **unverified**. That covers training and fine-tuning on Llama, Mistral, Qwen and Gemma-class models — the majority of what zerfoo supports. This is not a claim that training is broken; it is a claim that we have no evidence it is correct, and one skipped test saying it may not be.

## Suggested fix

1. Replace `gradClose` and its four siblings with a shared helper using a **combined absolute + relative** bound sized to measured error, per the pattern established in `docs/kernel-tolerances.md` for T135.3. Do not floor the denominator.
2. Re-run each suite against the corrected bound and record what actually fails. Expect real failures — that is the point.
3. Resolve the skipped `TestGQABackward`: either fix the analytical gradient or document precisely which configurations are affected and annotate them.
4. Every new bound must be red-proofed against a deliberately broken backward pass before it is trusted.

## Related

Same family as ztensor#182 (Q5_K bound ~1000x looser than measured error) and lore L-0009 (saturated fixtures that are blind to what they assert). Third distinct instance of this class.
